### PR TITLE
Added coinbase tagging of UTXOS

### PR DESCRIPTION
**Pull request by chiguireitor**

### CODE REVIEW

File: indexes/txin.js [hunk 1]
```diff
@@ -54,10 +56,16 @@ TxinIndex.prototype.connect = function (atomic, block, events) {
...
     ins.forEach((input, vin) => {
-      if (input.coinbase) return
+      let coinbase = ('coinbase' in input)?1:0
 
       let { prevTxId, vout } = input
-      atomic.put(TXIN, { txId: prevTxId, vout }, { txId, vin })
+
+      if (!prevTxId) {
+        prevTxId = '0000000000000000000000000000000000000000000000000000000000000000'
+        vout = 0
```
Comment (junderw) [inline indexes/txin.js]: iirc, coinbase inputs use 0xffffffff as vout.

File: indexes/txo.js [hunk 2]
```diff
@@ -74,7 +78,7 @@ TxoIndex.prototype.txoBy = function (db, txo, callback) {
   let { txId, vout } = txo
   let mem = this.txos[`${txId}:${vout}`]
   if (mem) return callback(null, mem)
-
+console.log('*****', txo)
```
Comment (junderw) [inline indexes/txo.js]: please remove

**Review by junderw — CHANGES_REQUESTED** [review]
Initial review. Request changes.

File: indexes/txo.js (hunk 2 above)
Comment (chiguireitor) [reply indexes/txo.js]: Doh!

**Comment by chiguireitor** [comment]
Changes in, please review as this inhibits any mining node to use indexd correctly

File: indexes/txin.js (hunk 1 above)
Comment (junderw) [reply indexes/txin.js]: This is what I meant.
```javascript
prevTxId = '0000000000000000000000000000000000000000000000000000000000000000'
vout = 0xffffffff
```

**Review by junderw — DISMISSED** [review]
please change vout, not txid. txid was fine as is. Thanks.

**Comment by chiguireitor** [comment]
Corrections in, check it out plz

**Comment by chiguireitor** [comment]
Hi, any progress on this? This is a breaking issue for regtest (which is a real good tool/timesaver for development)

**Comment by junderw** [comment]
While I have permissions to merge this into Github, I don't have publish rights to npm.

Also, I am not familiar with this repository or how it works, and there are no tests... so it is hard for me to evaluate whether this PR is OK.

@dcousens any thoughts?

**Comment by chiguireitor** [comment]
Aah yeah, npm is a deal breaker, as this goes into a lot of docker-compose stacks

**Review by abunsen — APPROVED** [review]
lgtm let's ship this puppy @dcousens

**Comment by junderw** [comment]
Looks like I was made into publisher on npm.

**Comment by junderw** [comment]
published as 0.9.1

**Pull request merged**